src/uproot/behaviors/TH1.py

```diff
@@ -109,7 +109,12 @@ def weighted(self):
         """
         True if the histogram has weights (``fSumw2``); False otherwise.
         """
-        raise NotImplementedError(repr(self))
+        sumw2 = self.member("fSumw2")
+        return (
+            sumw2 is not None
+            and len(sumw2) > 0
```

Comment:
This is the only addition after the initial review, I added it to make sure empty histograms are treated as non weighted. Probably it makes no difference. Let me know if you want this changed.

Reply:
I don't even know if zero-bin histograms are allowed in ROOT. If there is no bin data, I suppose it can't matter whether it's weighted or not, though I guess there's the underflow and overflow bins.